plugins/modules/system/sap_snote.py

```diff
@@ -0,0 +1,253 @@
+        description: The required password for the SAP system.
+        required: true
+        type: str
+    host:
+        description: The required host for the SAP system. Can be either an FQDN or IP Address.
+        required: true
+        type: str
+    sysnr:
+        description:
+        - The system number of the SAP system.
+        - You must quote the value to ensure retaining the leading zeros.
+        required: false
+        default: '01'
+        type: str
+    client:
+        description:
+        - The client number to connect to.
+        - You must quote the value to ensure retaining the leading zeros.
+        required: false
+        default: '000'
+        type: str
+    snote_path:
+        description:
+        - The path to the extracted SNOTE txt file.
+        - The File could be extracted from SAR package.
+        - If C(snote_path) is not provided, the C(snote) parameter must be defined.
+        - The SNOTE txt file must be at a place where the SAP System is authorized for. For example C(/usr/sap/trans/files).
+        required: false
+        type: str
+    snote:
+        description:
+        - With the C(snote) paramter only implementation and deimplementation will work.
+        - Upload SNOTES to the System is only available if C(snote_path) is provided.
+        required: false
+        type: str
+
+requirements:
+    - pyrfc >= 2.4.0
+
+author:
+    - Rainer Leber (@rainerleber)
+'''
+
+EXAMPLES = r'''
+- name: test snote module
+  hosts: localhost
+  tasks:
+  - name: implement SNOTE
+    community.sap.sap_snote:
+      conn_username: 'DDIC'
+      conn_password: 'Passwd1234'
+      host: 100.0.201.20
+      sysnr: '01'
+      client: '000'
+      state: present
+      snote_path: /usr/sap/trans/tmp/0002949148.txt
+
+- name: test snote module without path
+  hosts: localhost
+  tasks:
+  - name: deimplement SNOTE
+    community.sap.sap_snote:
+      conn_username: 'DDIC'
+      conn_password: 'Passwd1234'
+      host: 100.0.201.20
+      sysnr: '01'
+      client: '000'
+      state: absent
+      snote: 0002949148
+
+'''
+
+RETURN = r'''
+msg:
+    description: A small execution description.
+    type: str
+    returned: always
+    sample: 'SNOTE 000298026 implemented.'
+out:
+    description: A complete description of the SNOTE implementation. If this is available.
+    type: list
+    elements: dict
+    returned: always
+    sample: '{
+        "RETURN": [{"ES_MSG": { "MSGNO": "000", "MSGTY": "", "MSGTXT": "", "MSGV1": "" },
+                "ET_MSG": [],
+                "EV_RC": 0,
+                "ET_MISSING_NOTES": [],
+                "IT_FILENAME": [{"FILENAME": "/usr/sap/trans/tmp/0002980265.txt"}],
+                "IT_NOTES": [{"NUMM": "0002980265", "VERSNO": "0000"}]
+                }]}'
+'''
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+import traceback
+try:
+    from pyrfc import Connection
+except ImportError:
+    HAS_PYRFC_LIBRARY = False
```

Comment:
I guess this global (or the one later) comes from the (previous) import of Connection from pyrfc?

Reply:
as discussed